api/src/test/java/org/pmiops/workbench/cohortbuilder/CohortBuilderServiceImplTest.java

```diff
@@ -0,0 +1,133 @@
+package org.pmiops.workbench.cohortbuilder;
+
+import static com.google.common.truth.Truth.assertWithMessage;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.when;
+
+import java.time.Instant;
+import java.time.ZoneId;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Stream;
+import javax.inject.Provider;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.Mock;
+import org.pmiops.workbench.FakeClockConfiguration;
+import org.pmiops.workbench.api.BigQueryService;
+import org.pmiops.workbench.cdr.cache.MySQLStopWords;
+import org.pmiops.workbench.cdr.dao.CBCriteriaAttributeDao;
+import org.pmiops.workbench.cdr.dao.CBCriteriaDao;
+import org.pmiops.workbench.cdr.dao.CBDataFilterDao;
+import org.pmiops.workbench.cdr.dao.CriteriaMenuDao;
+import org.pmiops.workbench.cdr.dao.DomainCardDao;
+import org.pmiops.workbench.cdr.dao.PersonDao;
+import org.pmiops.workbench.cdr.dao.SurveyModuleDao;
+import org.pmiops.workbench.cohortbuilder.mapper.CohortBuilderMapper;
+import org.pmiops.workbench.cohortbuilder.mapper.CohortBuilderMapperImpl;
+import org.pmiops.workbench.test.FakeClock;
+import org.pmiops.workbench.workspaces.WorkspaceAuthService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.boot.test.context.TestConfiguration;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.context.annotation.Import;
+import org.springframework.test.annotation.DirtiesContext;
+
+@DataJpaTest
+@DirtiesContext(classMode = DirtiesContext.ClassMode.BEFORE_EACH_TEST_METHOD)
+class CohortBuilderServiceImplTest {
+
+  private static final Instant NOW = Instant.now();
+  private static final FakeClock CLOCK = new FakeClock(NOW, ZoneId.systemDefault());
+
+  @Mock private BigQueryService bigQueryService;
+  @Mock private CohortQueryBuilder cohortQueryBuilder;
+  @Autowired private CBCriteriaAttributeDao cbCriteriaAttributeDao;
+  @Autowired private CBCriteriaDao cbCriteriaDao;
+  @Autowired private CriteriaMenuDao criteriaMenuDao;
+  @Autowired private CBDataFilterDao cbDataFilterDao;
+  @Autowired private DomainCardDao domainCardDao;
+  @Autowired private PersonDao personDao;
+  @Autowired private SurveyModuleDao surveyModuleDao;
+  @Autowired private CohortBuilderMapper cohortBuilderMapper;
+  @Mock private Provider<MySQLStopWords> mySQLStopWordsProvider;
+
+  @TestConfiguration
+  @Import({FakeClockConfiguration.class, CohortBuilderMapperImpl.class})
+  @MockBean({WorkspaceAuthService.class})
+  static class Configuration {}
+
+  private CohortBuilderServiceImpl cohortBuilderService;
+
+  @BeforeEach
+  public void setUp() {
+
+    cohortBuilderService =
```

Comment:
You could just Autowire this service and add all your mock daos into the TestConfiguration

Reply:
yes, pl check the comment for #40